client/my-sites/promote-post-i2/components/campaign-item-details/index.tsx

```diff
@@ -370,6 +277,99 @@ export default function CampaignItemDetails( props: Props ) {
 		: '-';
 
 	const activeDays = getCampaignActiveDays( start_date, end_date );
+
+	const [ selectedDateRange, setSelectedDateRange ] = useState< ChartSourceDateRanges >(
+		activeDays >= 7 ? ChartSourceDateRanges.LAST_7_DAYS : ChartSourceDateRanges.WHOLE_CAMPAIGN
```

Comment:
this is the only line of code that has changed, the rest of the code is just moved with this so that the correct order is maintained